### DISCUSSION TITLE
Error when requesting to api "Error Page with `dynamic = "error""

### DISCUSSION THREAD

**Discussion by SabrioBiblio** [vercel/next.js#62601] [Help] [answered]
### Summary

I'm using **next js 14.0.4**.
Next config
```ts
const nextConfig = {
  ...(process.env.NODE_ENV = "production" ? { output: "export" } : {}),
  images: {
    unoptimized: true,
  },
  reactStrictMode: true,
  webpack(config) {
    config.module.rules.push({
      test: /\.svg$/,
      use: [{ loader: "@svgr/webpack", options: { icon: true } }],
    });
    return config;
  },
};
```
There is a main page with a client form component.
This form accesses **api/send-quote**.

```ts
export async function POST(req: Request) {
   const formData: FormData = await req.json();

   try {
     resend.emails.send({
       from: "...",
       to: [
         process.env.NODE_ENV === "production"
...
       ],
       subject: "New quote!",
       html:`
       <div>
         <h1>You have a new qoute!</h1>
         <br/>
         <p>From: ${formData.data.from}</p>
         <p>To: ${formData.data.to}</p>
         <p>Phone: ${formData.data.phone}</p>
       </div>`,
     });
     return Response.json(1);
   } catch (error) {
     return Response.json({ error });
   }
}
```

I'm getting an error in the development environment "npm run dev"
**Page with `dynamic = "error"` couldn't be rendered statically because it used `request.json`.**
And I get a 405 error in the production environment.
The site was generated using npm run build and deployed on vercel

### Example

https://skyboxmoving.com/

**Comment by icyJoseph** [comment, marked as answer]
Hi,

Why are you using `export` mode for production? That mode produces static HTML, which means you get JS, CSS, and HTML that you can then take to a static assets server. Notably, it does not produce any server side code, that is, no next server is built.

That means there is not server to handle the incoming request to api/send-quote. Take a look at:

- https://nextjs.org/docs/app/building-your-application/deploying/static-exports#unsupported-features 

What Next can do with Route handlers these days, is try to statically calculate their output at build time, but that's only possible for Routes return always the same data. The moment you make this route dynamic, in the sense that the response depends on the request, in any fashion, the export mode can't support it.

**Reply by SabrioBiblio** [reply]
@icyJoseph Thank you, now it’s clearer to me why the request doesn’t work.

I decided to use static for better performance, since my posting system relies on reading post content from mdx files, for some reason it takes about three seconds to go to the post page. Statics does this instantly. I'm not too strong in the logic of how next js works yet, so I make such mistakes

**Reply by SabrioBiblio** [reply]
Is it possible to create static for certain pages? For example, the hompeage page will be as static, and the blog page as ssg

**Reply by icyJoseph** [reply]
Hi,

Well Next.js is always, aggressively some might say, trying to pre-render as static assets. 

If a page meets certain conditions, it'll be statically optimized. Let me try to find some docs for it. So if a page uses no `use client`, and doesn't need per request info, Next.js serves, ideally, only HTML for that page. I say ideally, because framework stuff is still shipped, but as far as code written by you, under these conditions, very little is sent to the client.

**Reply by icyJoseph** [reply]
Here: https://nextjs.org/docs/app/api-reference/file-conventions/route-segment-config#dynamic

This is said about the dynamic page property:

> 'auto' (default): The default option to cache as much as possible without preventing any components from opting into dynamic behavior.

Here you can read about the default optimizations made by Next.js:

- https://nextjs.org/docs/app/building-your-application/deploying/production-checklist#automatic-optimizations
- https://nextjs.org/docs/app/building-your-application/rendering/server-components#static-rendering-default

**Reply by icyJoseph** [reply]
And last, but not least, you can see in your build output, which routes are statically rendered for optimization:

```console
Route (app)                              Size     First Load JS
┌ ○ /                                    479 B          84.6 kB
└ ○ /_not-found                          885 B            85 kB
+ First Load JS shared by all            84.2 kB
  ├ chunks/69-cb03b32d4bfc7b30.js        28.9 kB
  ├ chunks/fd9d1056-cc48c28d170fddc2.js  53.4 kB
  └ other shared chunks (total)          1.86 kB


ƒ Middleware                             39.5 kB

○  (Static)  prerendered as static content
```

As you can see, in this app, the `/` path is prerendered as static content.

**Reply by SabrioBiblio** [reply]
Thanks a lot! I just followed a guide from the internet where they said when using generateStaticParams to use export "output". There is no need to do this, it only deploy static. If you remove export "output", then statics will still be generated for those pages that use static parameters.
```
○ /                                                                    5.57 kB         118 kB
├ ○ /_not-found 
├ ● /blog/[slug]                                                         1.26 kB        94.9 kB
├   ├ /blog/choose-the-right-neighborhood-for-you-in-nyc
├   ├ /blog/moving-home-in-new-york-expert-tips-for-a-smooth-transition
├   ├ /blog/the-perfect-move-with-pets-to-new-york
├   └ /blog/tips-for-a-stress-free-move
```

**Reply by elidakirigo** [reply]
would you mind sharing where you read from?.. I'm having the same problem months later
